server/controllers/article.controller.js

```diff
@@ -158,13 +158,65 @@ class ArticleController {
       },
       {
         model: Comment,
-        as: 'comments'
+        as: 'articlecomments'
       }]
     });
     if (articles.count === 0) {
       return Response(res, 400, 'There is no article in database');
     }
     return Response(res, 200, 'Articles successfully retrieved', { articles });
   }
+
+  /**
+    * @description gets a single article
+    * @param {object} req
+    * @param {object} res
+    * @returns {object} An article and it's associated comments
+    */
+  static async getOneArticle(req, res) {
+    const articleSlug = req.params.slug;
+    const isArticleExists = await Article.findOne({
+      where: { slug: articleSlug },
+      include: [
+        {
+          model: User,
+          as: 'author',
+          attributes: ['id', 'username', 'bio', 'image']
+        },
+        {
+          model: Comment,
+          as: 'articlecomments',
+          attributes: ['comment']
+        }
+      ],
+    });
+
+    if (!isArticleExists) return Response(res, 404, 'Article not found.');
+
+    const {
+      dataValues: {
+        id, slug, title, description, body, tags, author, articlecomments
+      }
+    } = isArticleExists;
+
+    const { authorization } = req.headers;
```

Comment:
There is a middleware available on the code base to decode the token and get the user ID, can you make use of the middleware, or is there another reason you used this method?

Reply:
I didn't use the middleware because the route is supposed to be available for both logged in and unregistered users.

Reply:
True, but the middleware does exactly the same thing, so if its not meant to be protected then the authorization limits should not be there

Reply:
I need the user's id (if the user is logged in). Thats why the authorization stuff is there. It will only be triggered if the user is logged in.